engines/director/lingo/xlibs/mmovie.cpp

```diff
@@ -133,25 +239,395 @@ void MMovieXObj::m_new(int nargs) {
 
 XOBJSTUB(MMovieXObj::m_Movie, 0)
 XOBJSTUBNR(MMovieXObj::m_dispose)
-XOBJSTUB(MMovieXObj::m_openMMovie, 0)
-XOBJSTUB(MMovieXObj::m_closeMMovie, 0)
-XOBJSTUB(MMovieXObj::m_playSegment, 0)
-XOBJSTUB(MMovieXObj::m_playSegLoop, 0)
-XOBJSTUB(MMovieXObj::m_idleSegment, 0)
-XOBJSTUB(MMovieXObj::m_stopSegment, 0)
-XOBJSTUB(MMovieXObj::m_seekSegment, 0)
+
+void MMovieXObj::m_openMMovie(int nargs) {
+	g_lingo->printSTUBWithArglist("MMovieXObj::m_openMMovie", nargs);
+	if (nargs != 1) {
+		g_lingo->dropStack(nargs);
+		g_lingo->push(Datum(-1));
+		return;
+	}
+	MMovieXObject *me = static_cast<MMovieXObject *>(g_lingo->_state->me.u.obj);
+	Common::String basename = g_lingo->pop().asString();
+	Common::Path path = findPath(basename);
+	if (path.empty()) {
+		g_lingo->push(MMovieError::MMOVIE_INVALID_OFFSETS_FILE);
+		return;
+	}
+	Common::Path offsetsPath = findPath(basename.substr(0, basename.size()-4) + ".ofs");
+	if (offsetsPath.empty()) {
+		g_lingo->push(MMovieError::MMOVIE_INVALID_OFFSETS_FILE);
+		return;
+	}
+	if (me->_moviePathMap.contains(basename)) {
+		g_lingo->push(MMovieError::MMOVIE_MOVIE_ALREADY_OPEN);
+		return;
+	}
+	Common::File offsetsFile;
+	if (!offsetsFile.open(offsetsPath)) {
+		g_lingo->push(MMovieError::MMOVIE_INVALID_OFFSETS_FILE);
+		return;
+	}
+
+	MMovieFile movie(path);
+	movie._video = new Video::QuickTimeDecoder();
+	if (!movie._video->loadFile(path)) {
+		warning("MMovieXObj::m_openMMovie(): unable to open QT file %s", path.toString().c_str());
+		delete movie._video;
+		movie._video = nullptr;
+	}
+	uint32 offsetCount = offsetsFile.readUint32BE();
+	offsetsFile.skip(0x3c);
+	debugC(5, kDebugXObj, "MMovieXObj:m_openMMovie(): opening movie %s (index %d)", path.toString().c_str(), me->_lastIndex);
+	for (uint32 i = 0; i < offsetCount; i++) {
+		Common::String name = offsetsFile.readString(0x20, 0x10);
+		uint32 start = offsetsFile.readUint32BE();
+		uint32 length = offsetsFile.readUint32BE();
+		debugC(5, kDebugXObj, "MMovieXObj:m_openMMovie(): adding segment %s (index %d): start %d (%dms) length %d (%dms)", name.c_str(), movie.segments.size(), start, Audio::Timestamp(0, start, movie._video->getTimeScale()).msecs(), length, Audio::Timestamp(0, length, movie._video->getTimeScale()).msecs());
+		movie.segments.push_back(MMovieSegment(name, start, length));
+		movie.segLookup.setVal(name, movie.segments.size());
+	}
+	me->_movies.setVal(me->_lastIndex, movie);
+	me->_moviePathMap.setVal(basename, me->_lastIndex);
+	g_lingo->push(me->_lastIndex);
+	me->_lastIndex += 1;
+}
+
+void MMovieXObj::m_closeMMovie(int nargs) {
+	g_lingo->printSTUBWithArglist("MMovieXObj::m_closeMMovie", nargs);
```

Comment:
Is it still a stub? The same question is for other methods. If you want to keep it for debugging purposes, I'd recommend creating another method, like `g_lingo->printArglist(...)`, and make `printSTUBWithArglist()` a wrapper for it, for reusability.